server/org.eclipse.emfcloud.ecore.glsp/src/main/java/org/eclipse/emfcloud/ecore/glsp/ResourceManager.java

```diff
@@ -112,9 +122,24 @@ public void save() {
 		try {
 			ecoreFacade.getSemanticResource().save(Collections.EMPTY_MAP);
 			ecoreFacade.getNotationResource().save(Collections.EMPTY_MAP);
+			handleSaveErrors(ecoreFacade.getSemanticResource());
+			handleSaveErrors(ecoreFacade.getNotationResource());
 		} catch (IOException e) {
 			throw new GLSPServerException("Could not save notation resource", e);
 		}
 	}
 
+	private void handleSaveErrors(Resource resource) {
+		if (resource.getErrors().isEmpty()) {
+			return;
+		}
+		
+		System.err.println("Some errors have been found while saving "+resource.getURI().lastSegment()+":");
```

Comment:
I feel like this should be logged with the log4j logger and we should throw  a GLSPServerException. This way the client will get notified as well

Reply:
Can we still return/trigger Actions when throwing an exception? Typically, I'd like the SetDirtyStateAction to be processed normally, even if some exceptions/warnings were reported during the save. Throwing an Exception will not allow me to return this SetDirtyStateAction